### PR TITLE
feat: include deprecation message in docs

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/terraform-provider-coder/issues/237

This PR adds a script to patch Terraform docs to contain the deprecation message.

### CODE REVIEW

File: scripts/docsgen/main.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	"strings"
+
+	"github.com/coder/terraform-provider-coder/provider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// This script patches Markdown docs generated by `terraform-plugin-docs` to expose the original deprecation message.
+
+var (
+	reDeprecatedProperty = regexp.MustCompile("`([^`]+)` \\(([^,\\)]+), Deprecated\\) ([^\n]+)")
```
Comment (johnstcn) [inline scripts/docsgen/main.go:19]: readability: use backticks for regexes so you don't need to escape all the things

File: scripts/docsgen/main.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func writeDeprecationMessage(doc []byte, schemas map[string]*schema.Schema) []byte {
+	return reDeprecatedProperty.ReplaceAllFunc(doc, func(m []byte) []byte {
+		matches := reDeprecatedProperty.FindSubmatch(m)
+		propertyName := matches[1]
+		description := matches[3]
+
+		sch := schemas[string(propertyName)]
+		if string(description) != sch.Description {
```
Comment (johnstcn) [inline scripts/docsgen/main.go:83]: nit: probably no harm to drop a log here for troubleshooting

File: scripts/docsgen/main.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func exposeDeprecationMessage(p *schema.Provider) error {
+	// Patch data-sources
+	for dataSourceName, dataSource := range p.DataSourcesMap {
+		docFile := filepath.Join("docs", "data-sources", strings.Replace(dataSourceName, "coder_", "", 1)+".md")
+
+		err := adjustDocFile(docFile, dataSource.Schema)
+		if err != nil {
+			return err
```
Comment (johnstcn) [inline scripts/docsgen/main.go:38]: wrap error

**Review by johnstcn — APPROVED** [review]
Thanks for figuring this out! 
Just a couple of small changes I'd like to see in the docsgen script to make it easier to debug changes in future.

File: scripts/docsgen/main.go (hunk 1 above)
Comment (mtojek) [reply scripts/docsgen/main.go:19]: no so easy here? we have backticks in the regexp already, and we can't escape them

Comment (johnstcn) [reply scripts/docsgen/main.go:19]: 🤦 never mind then!

File: scripts/docsgen/main.go (hunk 2 above)
Comment (mtojek) [reply scripts/docsgen/main.go:84]: Fixed

File: scripts/docsgen/main.go (hunk 3 above)
Comment (mtojek) [reply scripts/docsgen/main.go]: Fixed

**Pull request merged**